docs/2_3_common_states.adoc

```diff
@@ -82,485 +87,644 @@ If `loggingOn == fmi3True`, the FMU enables a vendor defined set of `<LogCategor
-An event occurs at this time instant, if indicated by the return argument `enterEventMode == fmi3True`.
-Such an event is called <<step event>>.
-_[<<step event,`Step events`>> are, for example, used to dynamically change the (continuous) <<state,`states`>> of a model internally in the FMU, because the previous states are no longer suited numerically.]_
+There are multiple kinds of events that require a transition to <<EventMode>>:
+
+* <<time-event,Time events>>: The time instant of such an event is signaled to the importer by
+  - the argument <<nextEventTime>> of <<fmi3UpdateDiscreteStates>>, or
+  - the arguments of <<fmi3GetShift>> and <<fmi3GetInterval>>.
+* <<state-event,State events>>
+* <<input-event,Input events>>
+* <<step-event,Step events>>
+* Co-Simulation: <<fmi3DoStep>> returns with <<eventEncountered,`eventEncountered == fmi3True`>> or <<clocksAboutToTick,`clocksAboutToTick == fmi3True`>>.
+
+[#table-math-event-mode]
+[cols="5,3",options="header",]
+|====
+|Equations and Actions
+|Functions Influencing Equations
+
+|Set <<tunable>> <<parameter,`parameters`>> latexmath:[\mathbf{p}_{\mathit{variability == tunable}}]
+|`fmi3Set{VariableType}`
+
+|Set continuous-time and discrete-time <<input,`inputs`>> latexmath:[\mathbf{u}_{c+d}(\mathbf{t})]
+|`fmi3Set{VariableType}`
 
-In this mode all continuous-time and discrete-time equations are active and the unknowns at an event can be computed and retrieved.
-The event time of a <<state event>> may be determined if a domain change of at least one event indicator is detected at the end of a completed integrator step.
+a|
+* Activate clocks and respective model partitions in latexmath:[\mathbf{f}_{\mathit{event}}]
+* Deactivate clocks and
+  - Reset respective clocked variables to their previous values: latexmath:[\mathbf{v}_{\mathit{d}} := {}^\bullet\mathbf{v}_{\mathit{d}}]
+  - Deactivate respective model partitions in latexmath:[\mathbf{f}_{\mathit{event}}]
+|<<fmi3SetClock>>
+
+a|
+* Activate output clocks and respective model partitions in latexmath:[\mathbf{f}_{\mathit{event}}] +
+* latexmath:[(\mathbf{y}_{c+d}, \mathbf{\dot{x}}_c, \mathbf{x}_{c+d}, \mathbf{z}, \mathbf{r}, \mathbf{w}_{c+d}, \mathbf{b}, \mathbf{T}_{\mathit{next}}, \mathbf{T}_\mathit{interval}) := \mathbf{f}_{\mathit{event}}({}^\bullet\mathbf{x}_{c+d}, \mathbf{u}_{c+d}, \mathbf{p}, {}^\bullet\mathbf{b}, \mathbf{t})]
+|<<fmi3GetClock>>, +
+<<fmi3GetInterval>>, +
+`fmi3Get{VariableType}`, +
+<<fmi3GetDerivatives>>, +
+<<fmi3GetContinuousStates>>, +
+<<fmi3GetEventIndicators>> +
+
+a|
+* Evaluate latexmath:[\mathbf{f}_{\mathit{event}}], if no `fmi3GetXXX` function was called in this instant of super-dense time
+* Update previous values of discrete states: latexmath:[{}^\bullet\mathbf{x}_d:=\mathbf{x}_d]
+* Update previous values of <<buffers>>: latexmath:[{}^\bullet\mathbf{b}:=\mathbf{b}]
+* Deactivate active clocks and respective model partitions in latexmath:[\mathbf{f}_{\mathit{event}}]
```

Comment:
Is this done automatically by calling fmi3UpdateDiscreteStates, or is it necessary to call something?

Reply:
This is the definition of what the function does that is on the right column.
This is the mathematical description of the function´s behaviour/action.